### DISCUSSION TITLE
Ping Targets

### DISCUSSION THREAD

**Discussion by Blazerfi** [wifinigel/wiperf#47] [Q&A] [answered]
Hello Nigel! Thank you so much for creating this awesome project!

 I am wondering if it is possible to add more than 5 ping targets in the config.ini file? When I try to add a sixth target it doesn't show up in tail -f /var/log/wiperf_agent.log. I read through the documentation and didn't see anything about it being a fixed number of targets. Is there any way to work around this?
 
 Thanks!

**Comment by wifinigel** [comment, marked as answer]
Hi Lance,

The number of ping targets is currently fixed at five targets in the code.

I would need to update the code to support this. 

The main reason for implementing it this way was that in the config file, it's quite tricky to know how many targets to check for, as the data structure capabilities of the 'ini' file format are quite limited.

I'm pretty sure I can fix this in a future release, I just need to think of the best approach and add it to the poller code.

Just thinking about it quickly, I wonder if maybe a `number_of_targets` field would solve the issue:

```
[Ping_Test]
; yes = enabled, no = disabled
enabled: yes
;
; Number of ping targets
; number_of_targets: 7
;
; first host we'd like to ping
ping_host1: google.com
;
; second host we'd like to ping
ping_host2: cisco.com
;
;  ### ....more targets here ###
;
;
; seventh host we'd like to ping
ping_host7: blah.com
```

**Reply by Blazerfi** [reply]
Thank you for such a quick reply Nigel! That would be wonderful! 

I will be deploying some RPI's running Wiperf inside a warehouse. They will be used to monitor the wireless in a specific area where mobile fulfillment robots go to retrieve their orders from the cloud. My idea is to run the RPI as a fixed wireless client to use to compare any abnormalities with the robots, and being able to ping the robots would be a helpful way to determine any latency on the local wireless/ethernet as opposed to latency to our cloud services. 

With that being said, I am also looking to add an MTR feature on the RPI and have a Grafana dashboard for any issues routing to the cloud. That is something separate that I am working on for these particular RPI's.

**Reply by Blazerfi** [reply]
btw, how many ping targets do you think would be appropriate for Wiperf, and what would be considered too many? I wouldn't want to go overboard.

**Comment by wifinigel** [comment]
As it does them sequentially, it would be as many as you can fit in to your (5m) poll cycle (which would be quite a lot).

The only caveat is that you would also need to modify the report templates on Grafana yourself to cater for the extra ping tests that you use.

I'll have a look at the code and make sure there are no show-stoppers and get back to you soon.

Nigel.

**Reply by Blazerfi** [reply]
Wonderful! Thank you! Again, this freakin thing is amazing! I am really looking forward to deploying it and sharing the info.

**Comment by wifinigel** [comment]
Update was quite minor to get this working. New version (minor update) coming in the next couple of days with this new feature included (just need to do docs & install script):

![image](https://user-images.githubusercontent.com/19818404/105911632-45bb7600-6022-11eb-8945-21ae4b936196.png)

**Reply by Blazerfi** [reply]
Nice! That was fast!

**Comment by lealog** [comment]
nice improvement :) this new "feature" will also affect the HTTP test?

**Comment by wifinigel** [comment]
Uh-oh....scope creep ;-)

....let me have a look, Lol.

**Comment by wifinigel** [comment]
Yes, I'll remove hard-coded limits on ping, http, dns & smb tests. I have the code running OK, just need to document & create the install script & update docs.

**Comment by lealog** [comment]
Thanks @wifinigel :)

**Comment by wifinigel** [comment]
OK guys, here it is: https://wifinigel.github.io/wiperf/whats_new_v2.2/

cc: @Blazerfi, @lealog

**Reply by Blazerfi** [reply]
Thank you so much!

**Reply by lealog** [reply]
Thanks  @wifinigel 
Just a remark on the documentation - https://wifinigel.github.io/wiperf/whats_new_v2.2/

You mention v2.3 but I think this is a typo. It should be v2.2, right?

> In **v2.3**, these limits are removed and the number of test targets is now user defined.

**Reply by wifinigel** [reply]
Oops! Thanks - fixed.
